deltachat-ffi/deltachat.h

```diff
@@ -4678,6 +4709,38 @@ void            dc_msg_set_override_sender_name(dc_msg_t* msg, const char* name)
 void            dc_msg_set_file               (dc_msg_t* msg, const char* file, const char* filemime);
 
 
+/**
+ * Mark Webxdc message shipped with the main app as a default integration.
+ * Example:
+ *
+ * ~~~
+ * dc_msg_t* msg = dc_msg_new(context, DC_MSG_WEBXDC);
+ * dc_msg_set_file(msg, path_to_maps_xdc);
+ * dc_msg_set_default_webxdc_integration(msg);
+ * dc_send_msg(dc_create_chat_by_contact_id(context, DC_CONTACT_ID_SELF), msg);
```

Comment:
> What happens in multi-device? Will second device receive non-hidden message?

this is unsolved, indeed. thanks for the pointer! in practise, there is no need to send the message out at all

> Or dc_msg_set_default_webxdc_integration can even accept file path directly, no need to create dc_msg_t etc.

that's then maybe the best solution, this would also hide the implementation details from the API.

Reply:
i skip sending out webxdc integrations over the wire. it is not needed and maybe even harmful - the integrations are currently always shipped with the main app.

(we still leave the way of building the message object on our own as it would (a) otherwise require quite some refactoring and, more important, (b) skipping sending out would be required anyways when we allow the user to replace the integration. using the same code paths seems a good approach here, esp. as these won't be that much used.  
and: things are still experimental, we need to see how these integrations work in general, things can also be changed easily)

Reply:
that should be doable. i hesitated a bit because i was not sure how android could set a file from the assets - but there is anyways only `setFile()` (no `setDataBytes()` or so yet :)

and if that's needed anyways for desktop, this is another argument. i'll add a commit